web/shared/src/typing_status.ts

```diff
@@ -195,3 +279,47 @@ export function update(
     const current_time = worker.get_current_time();
     actually_ping_server(worker, new_recipient, current_time, typing_started_wait_period);
 }
+
+export function update_editing_status(
+    edit_box_worker: EditingStatusWorker,
+    new_recipient: Recipient,
+    new_status: "start" | "stop",
+    typing_started_wait_period: number,
+    typing_stopped_wait_period: number,
+): void {
+    assert(new_recipient.notification_event_type === "typing_message_edit");
+    const message_id = new_recipient.id;
+    if (new_status === "start" && editing_state.has(message_id)) {
+        const state = editing_state.get(message_id)!;
+        state.idle_timer = start_or_extend_idle_timer_for_message_edit(
+            edit_box_worker,
+            message_id,
+            typing_stopped_wait_period,
+        );
+        editing_state.set(message_id, state);
+        maybe_ping_server_for_message_edit(edit_box_worker, message_id, typing_started_wait_period);
+        return;
+    }
+
+    if (new_status === "stop") {
+        stop_notification_for_message_edit(edit_box_worker, message_id);
+        return;
+    }
```

Comment:
Doing this case first make it easier to read, since then the above conditional can be just `if (editing_state.has(message_id)) {`